asciidoc/volume3/biceps-extension-provisions/tf3-ch-8.3.2.9.8-extension-timestamp.adoc

```diff
@@ -0,0 +1,105 @@
+
+* epoch versioning,
+* optional epoch time-step offsets,
+* optional versioning of `pm:CalibrationInfo/@Time`, `pm:AlertSystemState/@LastSelfCheck`, `pm:AlertConditionState/@DeterminationTime`, `pm:AbstractMetricValue/@StartTime`, `pm:AbstractMetricValue/@StopTime`, `pm:AbstractMetricValue/@DeterminationTime`, `pm:AbstractContextState/@BindingStartTime` and `pm:AbstractContextState/@BindingEndTime`. 
+
+[sdpi_level=+1]
+====== Model
+
+The clock epoch schema is available in <<vol3_appendix_a_xml_schemas_timestamp_version>>. <<vol3_example_extension_clock_discontinuities>> shows an exemplary XML instance of a <<vol2_clause_dev_30_message_getmdibresponse, {var_label_dev_30_message_getmdibresponse}>> from a device that has experienced two recent non-slewing time adjustments following three adjustments some time in the past. Of particular note:
+
+* the clock state includes epoch time-step offsets for epochs 3 and 4; earlier versions are not referenced and therefore not required,
+* the state for metric `m1` references epoch version 3; all timestamps in this state are versioned,
+* the timestamp for metric `m2` is not versioned; its timestamp is less than `pm:ClockState/@LastSet` and its value should be treated with greater suspicion than later timestamps, 
+* although the current time (`pm:ClockState/@DateAndTime`) is also less than `pm:ClockState/@LastSet`, the current time is always reported using the current time-reference frame; its value need not be treated with any more suspicion than normal,
+* each `sdpi:Epoch` includes a `@Version`, `@Timestamp` and `@Offset`; the timestamp is in the time-reference frame of the versioned epoch, adding the `@Timestamp` and `@Offset` provides a timestamp for an equivalent point in time for the next epoch version (see illustration below),
+* the default value of any timestamp not specifically versioned is the current epoch version. 
+
+`sdpi:Epoch/@Offset` gives the change in time from the `sdpi:Epoch/@Timestamp` in the previous epoch to an equivalent point in time in the new epoch: 
+
+image::vol3-diagram-biceps-ext-non-slewing_adj.svg[align=center]
+
+.Example MDIB state following two recent non-slewing time adjustments
+[#vol3_example_extension_clock_discontinuities]
+====
+[source,xml]
+----
+include::../../listings/vol3-clause-biceps-content-example-timestamp-version.xml[]
+----
+====
+
+[sdpi_level=+1]
+====== Requirements
+
+.R0600
+[sdpi_requirement#r0600,sdpi_req_level=shall]
+****
+The <<vol1_spec_sdpi_p_actor_somds_provider>> shall increment `sdpi:Epochs/@Version` by exactly one, beginning from 0, for every non-slewing time adjustment to any system clock used as part of its System Function Contribution (<<acronym_sfc>>). 
+
+****
+
+.R0601
+[sdpi_requirement#r0601,sdpi_req_level=shall]
+****
+A <<vol1_spec_sdpi_p_actor_somds_provider>> shall reset all versioned timestamps when it assigns a new MDIB sequence identifier (`pm:MdibVersionGroup/@SequenceId`). 
```

Comment:
"Reset" is a problematic term as there is no definiton of "reset" in XML instances. There needs to be explicit instruction to what has to be deleted, inserted, or updated.

Reply:
Yeah. I didn't know how to describe so I followed R0047 in 10207. 

How about: 
> A <<vol1_spec_sdpi_p_actor_somds_provider>> shall set the version of all versioned timestamps to 0 when it assigns a new MDIB sequence identifier (`pm:MdibVersionGroup/@SequenceId`).